### ISSUE TITLE
Feathers Auk Release

### ISSUE THREAD

**Issue by ekryski** [feathersjs/feathers#299]
# Feathers Auk

This is the release issue to keep track of all the moving parts across all the repos. Expected official release date Q4 2016.

The main focus of this release is to fix any major bugs, add some ancillary services, get auth hardened after battle testing, update the CLI and Generator to be more future proof, and add more documentation and examples.

![imgp0339](https://cloud.githubusercontent.com/assets/599019/14508889/dcc521d6-0186-11e6-919a-9d9723efe92c.jpg)
_the Auk giving you a very creepy, judgy stare_
## Core and Core Modules
- [x] [Support JS config files in addition to JSON in feathers-config](https://github.com/feathersjs/feathers-configuration/issues/12).
- [x] Consolidate common hooks into their own repo. Now [feathers-hooks-common](https://github.com/feathersjs/feathers-hooks-common)
- [x] [Bootstrap an app using JSON config files](https://github.com/feathersjs/feathers-bootstrap)
- [x] Support asynchronous initialization of services or whatever else. _accomplished via [feathers-bootstrap](https://github.com/feathersjs/feathers-bootstrap)_
- [x] "Live querying" via [feathers-reactive](https://github.com/feathersjs/feathers-reactive)
- [x] [Add `first`, `last`, `error` support to hooks](https://github.com/feathersjs/feathers-hooks/pull/110)
## Authentication
- [x] [Feathers Authentication 1.0](https://github.com/feathersjs/feathers-authentication/milestones/1.0)

## Service Adapters
- ~~Supporting special `$search` param on database adapters (https://github.com/feathersjs/feathers/issues/334)~~ _moved to Buzzard release_
- [x] [Mongoose 3.4](https://github.com/feathersjs/feathers-mongoose/milestones/3.4)
- [x] [RethinkDB](https://github.com/feathersjs/feathers-rethinkdb)
- [x] [Elasticsearch](https://github.com/feathersjs/feathers-elasticsearch)
- ~~Feathers Realm 1.0~~ _Removed. Realm is not stable enough for us yet._
- ~~[Feathers Stripe 1.0](https://github.com/feathersjs/feathers-stripe/milestones/1.0)~~
- ~~[Feathers Sendgrid 1.0](https://github.com/feathersjs/feathers-sendgrid/milestones/1.0)~~
- ~~[Feathers Twilio 1.0](https://github.com/feathersjs/feathers-twilio/milestones/1.0)~~ _moved to Buzzard release_
- ~~[Feathers Mailer 1.1](https://github.com/feathersjs/feathers-mailer/milestones/1.1)~~ _moved to Buzzard release_
- ~~[Feathers Apple Push Notifications 1.0](https://github.com/feathersjs/feathers-apple-push/milestones/1.0)~~ _No ETA_
- ~~[Feathers Google Push Notifications 1.0](https://github.com/feathersjs/feathers-google-push/milestones/1.0)~~ _No ETA_
## iOS & Android Clients
- ~~[Feathers Android 1.0](https://github.com/feathersjs/feathers-android/milestones/1.0)~~ 
- ~~[Feathers iOS 1.0](https://github.com/feathersjs/feathers-ios/milestones/1.0)~~

[EDIT] Some work has been started here but these are not a high priority for us at the moment. We'd ❤️  some help if someone wants to really own these and become part of the core team reply in a comment. We likely won't be officially looking to revisit them until some time in 2017.
## New CLI and Generator

**In progress**
- [x] [Feathers CLI 2.0](https://github.com/feathersjs/feathers-cli/milestones/2.0)
- [x] [Feathers Generator 1.0](https://github.com/feathersjs/feathers-generator/milestones/1.0)
## Documentation and Guides
- [x] [Vue.js chat guide](https://github.com/feathersjs/feathers-docs/pull/152) _couresty of @niallobrien_ 🎉 
- [ ] [Many-to-many Relationships with Sequelize](https://github.com/feathersjs/feathers-docs/issues/51) _@ekryski in progress_
- [x] [How to do social sign-in](https://github.com/feathersjs/feathers-docs/issues/60)
- [x] [How to sanitize and serialize data with hooks](https://github.com/feathersjs/feathers-docs/issues/134)
- [x] [Muli-part form data](https://github.com/feathersjs/feathers-docs/issues/111)
- [x] [Custom formatting errors](https://github.com/feathersjs/feathers-docs/issues/122)
- [ ] [Deploying to Heroku](https://github.com/feathersjs/feathers-docs/issues/66)  _@ekryski in progress_
- [ ] [Dockerizing a Feathers App](https://github.com/feathersjs/feathers-docs/issues/63) _@slajax in progress_
- [ ] [Clustered Microservices](https://github.com/feathersjs/feathers-docs/issues/70)
- [x] [Slim down and restructure to be more approachable, API centric and quicker to navigate](https://github.com/feathersjs/feathers-docs/issues/251)
- [ ] [Move guides to Medium](https://github.com/feathersjs/feathers-docs/issues/250)
## Examples
- [x] [Using Auth0 for authentication](https://github.com/feathersjs/feathers-demos/issues/10)
- [x] [How to do social sign-in](https://github.com/feathersjs/feathers-demos/issues/3)
- [x] [How to render server side templates](https://github.com/feathersjs/feathers-demos/issues/8)
- [ ] [Many-to-many Relationships with Sequelize](https://github.com/feathersjs/feathers-demos/issues/9) _in progress @ekryski_
- [x] [An Isomorphic React Starter](https://github.com/feathersjs/feathers-demos/issues/2)

**Comment by mcchrish** [comment]
> Feathers Apple Push Notifications 1.0
Feathers Google Push Notifications 1.0 

What's the estimates of these two services being ready?

**Comment by daffl** [comment]
There was some discussion in #254 but it does not seem as if any further work has been done so far. It might drop off the roadmap for this release since we still have a couple of higher priority items to tackle.

**Comment by thosakwe** [comment]
Will Twitter auth be added within Auk?

**Comment by marshallswain** [comment]
@thosakwe yep, @ekryski has worked Twitter OAuth into the release.  🎉

**Comment by celalo** [comment]
I am planning to start my new project using feathers. I'd like to start with auk release as `feathers-authentication-management` module (which is part of auk) offers quite some functionality that I'd like to make use of. 

I'd normally just `npm install` the prerelease but I could not figure it out how to do that when using `feathers-cli`. 

How can I start with auk release?

**Comment by eddyystop** [comment]
FYI celalo, `feathers-authentication-management` works with the current, pre-Auk version of `feathers` and `feathers-authentication` as well.

**Comment by celalo** [comment]
Brilliant! @eddyystop thank you for the info.

**Comment by lobosan** [comment]
Hi everyone, so excited to see the progress. I'm about to start a big project por a data-driven SPA so I was wondering if I should go with version 2 or wait till AUK is production ready. Any updates on the release date?

**Comment by daffl** [comment]
@lobosan You can create a new application by running

```
npm install feathers-cli@pre
mkdir testproject
cd testproject
feathers generate app
```

**Comment by bertho-zero** [comment]
@daffl You have forgotten -g for the feathers-cli installation, no?

**Comment by daffl** [comment]
Yes, indeed.

**Comment by ekryski** [comment]
Thanks for everyone's patience with this. 90% of Auk is already complete but a lot of the confusion is coming from the lack of docs. We are *very* close to being finished updating all the documentation.

After talking with @daffl here is an update (order matters) for what is left to finally get this sucker closed...

- [ ] Auth API docs
- [ ] Hook Docs
- [ ] Chat guide
- [ ] Merge to `auk` branch
- [ ] Finalize Feathers basics guide
- [ ] Any remaining generator fixes
- [ ] Release generator
- [ ] Merge feathers-chat updates
- [ ] Wrap up feathers-permissions
- [ ] Permissions API docs
- [ ] Finish feathers-client API docs
- [ ] Update Feathers RN chat app
- [ ] Move over/update any old/propsed guides into the guides section or as blog posts on medium
- [ ] RELEASE ANNOUNCEMENT!

It's been a long time coming and we've definitely had some scope creep (some justified and some not) but we've also fielded a lot of questions, helped a lot of people, fixed a ton of bugs and added some pretty awesome functionality.

**Comment by zusamann** [comment]
thank you very much for all your work on this! 😄

**Comment by rjulianialtimetrik** [comment]
Thanks a lot for your work! Really looking forward for that last, long wished, "RELEASE ANNOUNCEMENT!"
Way to go!

**Comment by ekryski** [comment]
We shipped the new docs earlier this week and have moved the old ones over to https://legacy.docs.feathersjs.com/.

A few sections of the guides just need a bit of updating but overall all the API docs are in place and up to date and all the modules that were pre-releases have been released officially!

**Comment by ekryski** [comment]
This is now officially closed! 💥 

You can read about the release announcement here: https://blog.feathersjs.com/the-auk-takes-flight-c4b3c070c6d.

We'll be migrating the old guides over to blog posts on Medium over the next while but in the mean time you can still access them at:

- https://legacy.docs.feathersjs.com/guides/readme.html
- https://legacy.docs.feathersjs.com/frameworks/readme.html

**Comment by thosakwe** [comment]
Congratulations!!!

**Comment by lobosan** [comment]
So awesome guys. Thanks for your great work.
I wish some day we can see nice JavaScript environment and platform designed to work well with major frontend frameworks. Something magic like what meteor was when just came out. In that platform I loved the things I was able to achieve with ease. I mean having only one place to go to learn what I need to do to have a real-time Enterprise ready in record time plus real-time and testing live server available with one command.

That was so awesome for a guy coming from PHP, MySQL, PostgreSQL, WordPress, JQuery, Joomla, ExtJS, Java, C, C++, Windows guy.

And if I can think of a more open, flexible and progressive platform, I would like to see something around open source technologies such us:

- FeathersJS AUK (Backend)
- VueJS, ReactJS, Polymer, Angular, Inferno (Frontend)
- Mobile first, Material Design, IOs, Windows, Web, Desktop (Universal)
- Offline, reactive, real-time, push notifications, SSR, SEO friendly, native, fast, secure
- Graphql, Rest APIs, Schema outside of databases (Schema)
- Database agnostic but giving priority to multi model databases such us ArangoDB or combinations of CauchDB​, MongoDB, Redis, Neo4J
- A building tool such us GraphCool with the ability to define the security layer over the Schema through an intelligent CLI that has staters ala VueJS to create different types of apps. 

In conclusion, my dream platform should allow me to create a Real time, Enterprise ready, PWA from day one with ease with one command and available in a web testing environment full of debugging tools, performance advice and security awareness

**Comment by eddyystop** [comment]
Meteor had US$ 30M to spend doing what they did (and still they seem to have pivoted to Apollo). Feathers has a core of 6 part-time people with no external funds, so we depend on the Feathers' community.

Perhaps you can start on one of the items you mentioned?

**Comment by thosakwe** [comment]
I agree with the above response. Implementing all the features you mentioned sounds like another year of mostly-thankless work from the Feathers team.

**Comment by myknbani** [comment]
Congratz and lots of thanks to the Feathers team!

With regards to the "dream platform" comment, too much magic gives you less control over parts you want to change.  Feather's modular approach is fine the way it is.  It does things "the Node way". 

😁 ✌️
